### ISSUE TITLE
missing integration roadmap

### ISSUE THREAD

**Issue by steadyclackin** [skycoin/cx#6]
We're interested in building distributed consensus applications on top of the blockchain but it's pretty hard to evaluate the sky tech stack not knowing the plan for how cx will be integrated into the sky ecosystem. Specific questions we have are:

* Will CX be able to consume and produce CXO?
* What is the planned execution model for CX? Obviously local execution will work but are there plans to support distributed execution with output consensus?
* (Related question) what's the roadmap for integrating CX with the blockchain

**Comment by amherag** [comment]
Yes. Integrating CX and CXO that they can work natively is very important.

We are working in integrating CX onto blockchain and showcasing Skyledger coins which have CX enabled as the scripting language for executing all transactions on the chain. However we have so many other higher project priorities now, that it has not been receiving as much development attention as we have wanted.

Distributed computation using multiple computers jointly running a CX program on blockchain will solve a lot of concurrency and distributed systems problems and we definitely plan to prototype system like this as a demo application. And in the long term Skycoin and Skywire will rely upon CX by decentralized computation and for coordinating the Skywire network.

I think we will start seeing applications in CX very soon but I could be six months or a year before we start seeing distributed computation demos in CX.

**Comment by steady-clackin** [comment]
I'm very happy this is the vision! I (and others) would like to build a communications platform that leverages consensus computing to be "server free". Would love to be able to build on skyledger with cx, and maybe help contribute to sky to get there faster. Perhaps my project can be the first demo?

**Comment by steady-clackin** [comment]
This is all just a slight generalization of transaction processing on the blockchain, right? I mean what exists today is already a special case of a consensus-based distributed computer. What are the blockers to integrating CX on the blockchain? I mean isn't it just a generalization of a specific part of the code to run some other function instead of the transaction processing function?

**Comment by amherag** [comment]
Skycoin is still written in golang and is not running on cx yet, but you could start writing blockchain projects in CX now.

**Comment by steady-clackin** [comment]
The cx interpreter is golang, so can't skycoin just invoke the interpreter to run the cx?

**Comment by steady-clackin** [comment]
> We are working in integrating CX onto blockchain and showcasing Skyledger coins which have CX enabled as the scripting language for executing all transactions on the chain. However we have so many other higher project priorities now, that it has not been receiving as much development attention as we have wanted.

How can I help?

**Comment by steady-clackin** [comment]
> And in the long term Skycoin and Skywire will rely upon CX by decentralized computation and for coordinating the Skywire network.

This is awesome btw!

**Comment by amherag** [comment]
We got a roadmap now! If you have any questions about it (look in the README of this repo), submit a new issue :)
